### ISSUE TITLE
Gitolite username mismatch

### ISSUE THREAD

**Issue by mlutfy** [jbox-web/redmine_git_hosting#460]
Context: I have a small Redmine instance with redmine_git_hosting that I have been upgrading since the early days of gitolite v1. Currently everything runs the latest versions of Redmine 3.x, redmine_git_hosting v1.1.x and gitolite on Debian Jessie 8.x.

Everything seems to work pretty well since the latest upgrade, but I sometimes have problems with some users. For example:

* I have a user in Redmine, cleaned out all the ssh keys, and re-added a key. Redmine displays it as: "keydir/redmine_git_hosting/redmine_joe_29/redmine_docuwiki/redmine_joe_29.pub"

* In my gitolite.conf, I see the following configuration that was created:

```
repo    myproject
  RW+                            = redmine_joe_29
  config redminegitolite.projectid = myproject
  config redminegitolite.repositoryid =
  config redminegitolite.repositorykey = [...]
```

* In the gitolite keydir, I see:
 * ~/gitolite/keydir/redmine_joe@redmine_1420827416_508769.pub
 * ~/gitolite/keydir/redmine_git_hosting/redmine_joe_29/redmine_joe/redmine_joe_29.pub

* However, when I test "ssh git@git.example.org info", I get:

```
hello redmine_joe, this is git@myserver running gitolite3 3.6.1-2 (Debian) on git 2.1.4

 R W    testing
```

In other words, I expected gitolite to respond with "hello redmine_joe_29". If I manually edit my gitolite.conf, to use the example below, it works:

```
repo    myproject
  RW+                            = redmine_joe
  config redminegitolite.projectid = myproject
  config redminegitolite.repositoryid =
  config redminegitolite.repositorykey = [...]
```

With some other projects, with other users, I do not seem to be having this problem.

I suspect old configurations causing problems. Any hints on what could be causing this?

Thanks,
Mathieu

**Comment by n-rodriguez** [comment]
Your SSH key is duplicated : 

```
~/gitolite/keydir/redmine_joe@redmine_1420827416_508769.pub
~/gitolite/keydir/redmine_git_hosting/redmine_joe_29/redmine_joe/redmine_joe_29.pub
```
Gitolite will pick the first one it'll find so here :  ```~/gitolite/keydir/redmine_joe@redmine_1420827416_508769.pub```.
You must remove this key by hand.

**Comment by mlutfy** [comment]
Many thanks for the quick response!

I cloned the gitolite-admin repo, removed the key from the keydir, pushed, then re-tested from Redmine to add/remove, and all worked OK.
